### PR TITLE
Add optional GitHub organisation authentication

### PR DESCRIPTION
**Pull request by manics**
This allows users authenticated with GitHub OAuth access if they are a member of a specified GitHub organisation.

WARNING: The molecule test is only a partial test of the role (it doesn't check that a notebook is running), and requires the vagrant driver, so can't be tested in travis.

### CODE REVIEW

**Comment by joshmoore** [comment]
> so can't be tested in travis.

So we should look to test within idr-ci?

**Comment by manics** [comment]
You'd need to get vagrant working in idr-ci. Manual testing is the most feasible solution at the moment.

File: .travis.yml [hunk 1]
```diff
@@ -1,29 +1,18 @@
...
 install:
-  # Install ansible
-  - pip install ansible
-
-  # Check ansible version
-  - ansible --version
-
-  # Create ansible.cfg with correct roles_path
-  - printf '[defaults]\nroles_path=../' >ansible.cfg
+  - pip install ansible docker 'molecule<1.22'
```
Comment (joshmoore) [inline .travis.yml:8]: Bump to the new package?

Comment (manics) [reply .travis.yml:8]: Done (looks like I forgot to enable `molecule test` in travis)

**Comment by joshmoore** [comment]
Failed re: vagrant.

**Comment by manics** [comment]
I've now remembered why I didn't update travis. I've forcibly removed the last commit.

Jupyterhub is running as Docker-in-Docker, so the molecule test involves Docker-in-(Docker-in-Docker) which doesn't work.

**Comment by joshmoore** [comment]
Fair enough. Merging and releasing `1.1.0` then.

**Pull request merged**